src/emitters/emitter.sqns.ts

```diff
@@ -37,761 +38,767 @@ import { SQSProducer } from "../producers/producer.sqs";
-      })
-    );
-    return result.Failed.map((failed) => ({
-      id: failed.Id,
-      code: failed.Code,
-      message: failed.Message,
-      wasSenderFault: failed.SenderFault,
-    }));
-  }
-
-  async emitBatch(
-    eventName: string,
-    messages: IBatchMessage[],
-    options?: IBatchEmitOptions
-  ): Promise<IFailedEmitBatchMessage[]> {
-    try {
-      const topic: Topic = {
-        name: eventName,
-        isFifo: !!options?.isFifo,
-        exchangeType: options?.exchangeType || ExchangeType.Fanout,
-        separateConsumerGroup: options?.consumerGroup,
-      };
-      if (topic.exchangeType === ExchangeType.Queue) {
-        return await this.emitBatchToQueue(topic, messages);
-      }
-      return await this.emitBatchToTopic(topic, messages);
-    } catch (error) {
-      Logger.error(
-        `Batch Message producing failed: ${eventName} ${JSON.stringify(error)}`
-      );
-      throw error;
-    }
-  }
-
-  async startConsumers() {
-    if (this.consumersStarted) {
-      return;
-    }
-    this.queues.forEach((queue) => {
-      if (!queue || queue.isDLQ || queue.listenerIsLambda) {
-        return;
-      }
-      this.startConsumer(queue);
-    });
-    this.consumersStarted = true;
-    Logger.info(`Consumers started`);
-  }
-
-  private startConsumer(queue: Queue) {
-    if (!queue.url) {
-      return;
-    }
-    queue.consumer = Consumer.create({
-      /**
-       * Handling delete message explcitly because sqs-consumer
-       * does not delete the successful ones if one of the message
-       * in the batch throws
-       */
-      shouldDeleteMessages: false,
-      region: this.options.awsConfig?.region,
-      queueUrl: queue.url,
+		}`;
+	};
+
+	private logFailedEvent = (data: IFailedEventMessage) => {
+		if (!this.options.eventOnFailure) {
+			return;
+		}
+		this.localEmitter.emit(this.options.eventOnFailure, data);
+	};
+
+	async emitToTopic(
+		topic: Topic,
+		options?: IEmitOptions,
+		...args: any[]
+	): Promise<boolean> {
+		const topicArn = this.getTopicArn(this.getTopicName(topic));
+		await this.snsProducer.send(
+			topicArn,
+			this.getTransformedMessage(topic, args, options)
+		);
+		return true;
+	}
+
+	async emitToQueue(
+		topic: Topic,
+		options?: IEmitOptions,
+		...args: any[]
+	): Promise<boolean> {
+		const queueUrl = this.getQueueUrl(this.getQueueName(topic));
+		await this.sqsProducer.send(
+			queueUrl,
+			this.getTransformedMessage(topic, args, options),
+			{
+				delay: options?.delay || DEFAULT_MESSAGE_DELAY,
+			}
+		);
+		return true;
+	}
+
+	getTransformedMessage(
+		topic: Topic,
+    args: any[],
+		options?: Pick<IEmitOptions, "partitionKey" | "MessageAttributes" | "deduplicationId">,
+	): IMessage {
+		const message: IMessage = {
+			messageGroupId: options?.partitionKey || topic.name,
+			eventName: topic.name,
+			data: args,
+			messageAttributes: options?.MessageAttributes,
+			deduplicationId: options?.deduplicationId,
+		};
+		return message;
+	}
```

Comment:
Is transformation same for both sqs and sns ? Did you check whether you need to stringify inside this function or not, you probably do ?